src/m365/pa/commands/app/app-get.ts

```diff
@@ -33,8 +34,17 @@ class PaAppGetCommand extends AzmgmtCommand {
       logger.logToStderr(`Retrieving information about Microsoft Power App ${args.options.name}...`);
     }
 
+    let requestUrl: string = '';
+    const isValidGuid: boolean = Utils.isValidGuid(args.options.name);
+    if (isValidGuid) {
+      requestUrl = `${this.resource}providers/Microsoft.PowerApps/apps/${encodeURIComponent(args.options.name)}?api-version=2016-11-01`
+    }
+    else {
+      requestUrl = `${this.resource}providers/Microsoft.PowerApps/apps?api-version=2016-11-01`
```

Comment:
Issuing this request just once won't work if you have a lot of Power App apps. If I'm not mistaken, this request returns up to 500 items at a time, so if we need to get all apps first, we should call the `pa app list` command internally and then loop through its results to find the right app.

Reply:
Here's one example: https://github.com/pnp/cli-microsoft365/blob/bbd07bae6f59d02d191b7b1472e39752895284cb/src/m365/spo/commands/site/site-ensure.ts#L136-L148